src/wms/src/main/java/org/geoserver/wms/decoration/LegendDecoration.java

```diff
@@ -101,6 +107,22 @@ public Dimension findOptimalSize(Graphics2D g2d, WMSMapContent mapContext) {
                 request.setKvp(dispatcherRequest.getKvp());
                 request.setRawKvp(dispatcherRequest.getRawKvp());
             }
+            // online resource handling
+            LayerInfo info = wms.getLayerByName(layer.getTitle());
+            StyleInfo defaultStyle = info.getDefaultStyle();
+            StyleInfo sInfo =
+                    info.getStyles()
+                            .stream()
+                            .filter(s -> s.getName().equals(layer.getStyle().getName()))
```

Comment:
GeoServer is prone to have dangling style references, best to filter out null styles in this loop.

Reply:
fixed